### PR TITLE
Prevent UCX from using the 0.32.{0,1} releases of the Databricks SDK.

### PR DESCRIPTION
**Pull request by asnare**
## Changes

This PR updates the project dependencies so that we don't use the 0.32.0 or 0.32.1 releases of the SDK. One of the [changes](/databricks/databricks-sdk-py/pull/741) in that release seems to have introduced some regressions in existing error-handling. Although we can work around some of them, others (such as handling of permission-denied errors) we can't.

Given that this particular version of the SDK didn't introduce new features/behaviour that we rely upon, the safest path seems to be ensure that we don't install it specifically.

### Linked issues

Relates: databricks/databricks-sdk-py#749, databricks/databricks-sdk-py#755, 

Fixes: #2542

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -44,7 +44,7 @@ classifiers = [
     "Topic :: Utilities",
 ]
 
-dependencies = ["databricks-sdk~=0.30",
+dependencies = ["databricks-sdk~=0.30,!=0.32.0",
```
Comment (JCZuurmond) [inline pyproject.toml:47]: Please add a comment explaining why we exclude this version

**Review by JCZuurmond — COMMENTED** [review]
See comment

**Comment by asnare** [comment]
The 0.32.1 release has been made, which partly addresses the error-handling regressions. It doesn't address #2542 though.

**Comment by asnare** [comment]
Version 0.32.2 of the SDK has been released, and it includes a fix for the remaining regression we noticed with error-handling. Rather than complicate the dependency-specification we'll just leave it as-is for now.

**Pull request closed without merging**